packages/plugins/scripts-runner/src/index.ts

```diff
@@ -0,0 +1,178 @@
+        const dependencies = await this.getScriptDependencies();
+        await this.executeAll(targetPath, dependencies, callback);
+      } else if (fstat.isFile()) {
+        const dependencies = await this.getScriptDependencies();
+        await this.executeSingle(targetPath, dependencies, callback);
+      } else {
+        callback(new UnsupportedTargetError(fstat));
+      }
+    } catch (e) {
+      callback(e);
+    }
+  }
+
+  private async executeSingle(target: string, dependencies: ScriptDependencies, callback: Callback<any>) {
+    const scriptToRun = require(target);
+    this.embark.logger.info(`.. ${path.basename(target)} running....`);
+    try {
+      const result = await scriptToRun(dependencies);
+      this.embark.logger.info(`.. finished.`);
+      callback(null, result);
+    } catch (e) {
+      const error = e instanceof Error ? e : new Error(e);
+      callback(new ScriptExecutionError(target, error));
+    }
+  }
+
+  private async executeAll(target: string, dependencies: ScriptDependencies, callback: Callback<any>) {
+    const files = await fsPromises.readdir(target);
+    const results: any[] = [];
+
+    eachSeries(files, (file, cb) => {
+      const targetPath = !path.isAbsolute(target) ? dappPath(target, file) : path.join(target, file);
+
+      stat(targetPath, (err, fstat) => {
+        if (err) {
+          return cb(err);
+        }
+        if (fstat.isFile()) {
+          return this.executeSingle(targetPath, dependencies, (e, result) => {
+            if (e) {
+              return cb(e);
+            }
+            results.push(result);
+            cb();
+          });
+        }
+        cb();
+      });
+    }, (err) => {
+      if (err) {
+        return callback(err);
+      }
+      callback(null, results);
+    });
+  }
+
+  private async getScriptDependencies() {
+    const contracts = await this.embark.events.request2('contracts:list');
+
+    const dependencies: ScriptDependencies  = {
+      logger:  this.embark.logger,
+      web3: null,
+      contracts: {}
+    };
+
+    dependencies.web3 = await this.web3;
+
+    for (const contract of contracts) {
+      const registeredInVM = this.checkContractRegisteredInVM(contract);
+      if (!registeredInVM) {
+        await this.embark.events.request2("embarkjs:contract:runInVm", contract);
+      }
+      const contractInstance = await this.embark.events.request2("runcode:eval", contract.className);
+      dependencies.contracts[contract.className] = contractInstance;
+    }
+    return dependencies;
+  }
+
+  // TODO(pascal): the same API is used in the specialconfigs plugin.
+  // Might make sense to either move this to code-runner and set up a
+  // command handler, or set up a command handler in specialconfigs.
+  private async checkContractRegisteredInVM(contract) {
+    const checkContract = `
+      return typeof ${contract.className} !== 'undefined';
+    `;
+    return this.embark.events.request2('runcode:eval', checkContract);
+  }
```

Comment:
Wasn't exactly sure if we want to depend on another plugin for that, so please let me know what you think

Reply:
Yeah, might be worth creating a command for that. Can be in another PR